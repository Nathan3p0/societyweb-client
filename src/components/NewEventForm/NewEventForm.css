.admin__events-form {
    width: 500px;
    max-width: 100%;
    padding: 1rem;
}

.admin__events-form h3 {
    text-align: center;
    padding: 1rem;
    background-color: #26495E;
    color: #fff;
}


.newEvent__form li {
    display: flex;
    flex-wrap: wrap;
    align-items: center;
    margin-top: .5rem
}

.newEvent__form > li > label {
    flex: 1 0 100px;
}

.newEvent__form > li > label + * {
    flex: 1 0 220px;
}

.newEvent__form li button {
    margin-left: auto;
    padding: 8px 16px;
    border: none;
    background: #333333;
    color: #f2f2f2;
    text-transform: uppercase;
    letter-spacing: .09em;
    border-radius: 2px;
}

.newEvent__form li button:hover {
    cursor: pointer;
    transform: scale(1.1);
    box-shadow: 1px 1px 5px #333333;
}

.newEvent__form li input {
    padding: 15px;
    border: none;
}