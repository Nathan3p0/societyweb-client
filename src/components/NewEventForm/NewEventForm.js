import React from 'react';
import './NewEventForm.css'

const NewEventForm = (props) => {
    return (
        <form onSubmit={props.handleSubmit} className="admin__events-form">
            <h3>Create a new event:</h3>
            <ul className="newEvent__form">
                <li>
                    <label htmlFor="event_name">Event Name:</label>
                    <input type="text" name="event_name" id="event_name" placeholder="Event Name" required />
                </li>
                <li>
                    <label htmlFor="event_time">Event Time:</label>
                    <input type="time" name="event_time" id="event_time" required />
                </li>
                <li>
                    <label htmlFor="event_date">Event Date:</label>
                    <input type="date" name="event_date" id="event_date" required />
                </li>
                <li>
                    <label htmlFor="event_location">Event Location:</label>
                    <input type="text" name="event_location" id="event_location" placeholder="Location" required />
                </li>
                <li>
                    <label htmlFor="event_description">Event Description:</label>
                    <textarea name="event_description" id="event_description" placeholder="Write description here..." required />
                </li>
                <li>
                    <button type="submit">Create Event</button>
                </li>
            </ul>
        </form>
    );
}

export default NewEventForm;