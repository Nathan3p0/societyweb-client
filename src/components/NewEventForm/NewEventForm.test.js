import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import NewEventForm from './NewEventForm'

describe('New Event Form Component Rendering Tests', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<NewEventForm />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders to UI as expected', () => {
        const tree = renderer.create(<NewEventForm />).toJSON();
        expect(tree).toMatchSnapshot()
    })
})