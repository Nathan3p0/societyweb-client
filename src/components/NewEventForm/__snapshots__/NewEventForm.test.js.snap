// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Event Form Component Rendering Tests renders to UI as expected 1`] = `
<form
  className="admin__events-form"
>
  <h3>
    Create a new event:
  </h3>
  <ul
    className="newEvent__form"
  >
    <li>
      <label
        htmlFor="event_name"
      >
        Event Name:
      </label>
      <input
        id="event_name"
        name="event_name"
        placeholder="event name"
        required={true}
        type="text"
      />
    </li>
    <li>
      <label
        htmlFor="event_time"
      >
        Event Time:
      </label>
      <input
        id="event_time"
        name="event_time"
        required={true}
        type="time"
      />
    </li>
    <li>
      <label
        htmlFor="event_date"
      >
        Event Date:
      </label>
      <input
        id="event_date"
        name="event_date"
        required={true}
        type="date"
      />
    </li>
    <li>
      <label
        htmlFor="event_location"
      >
        Event Location:
      </label>
      <input
        id="event_location"
        name="event_location"
        placeholder="location"
        required={true}
        type="text"
      />
    </li>
    <li>
      <label
        htmlFor="event_description"
      >
        Event Description:
      </label>
      <textarea
        id="event_description"
        name="event_description"
        placeholder="Write description here..."
        required={true}
      />
    </li>
    <li>
      <button
        type="submit"
      >
        Create Event
      </button>
    </li>
  </ul>
</form>
`;
